deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py

```diff
@@ -94,22 +103,258 @@ def get_data_sample(self) -> dict:
+                                        'ID': get_random_string(64),
+                                    },
+                                    'Grants': [
+                                        {
+                                            'Grantee': {
+                                                'Type': 'CanonicalUser',
+                                                'DisplayName': 'wazuh',
+                                                'ID': get_random_string(64),
+                                            },
+                                            'Permission': 'FULL_CONTROL'
+                                        },
+                                        {
+                                            'Grantee': {
+                                                'Type': 'Group',
+                                                'URI': 'http://acs.amazonaws.com/groups/global/AllUsers'
+                                            },
+                                            'Permission': 'READ'
+                                        }
+                                    ]
+                                }
+                            ]
+                        },
+                        'Event Count': 1,
+                        'Timestamps': {
+                            '2021-01-01T00:11:49.171020Z': 1
+                        },
+                        'recipientAccountId': {
+                            cons.RANDOM_ACCOUNT_ID: 1
+                        }
+                    },
+                    'trigger': {
+                        'rule-arn': (
+                            f"arn:aws:macie:us-east-1:{cons.RANDOM_ACCOUNT_ID}:trigger/b731d9ffb1fe61508d4a478c92efa666"
+                        ),
+                        'alert-type': 'basic',
+                        'created-at': '2020-12-29 16:36:17.412000+00:00',
+                        'description': 'S3 Bucket uses IAM policy to grant read rights to Everyone.',
+                        'risk': 9
+                    }
+                }
+            }
+        )
 
 
 # Maps bucket type with corresponding data generator
 buckets_data_mapping = {
-    cons.CLOUD_TRAIL_TYPE: CloudTrailDataGenerator
+    cons.CLOUD_TRAIL_TYPE: CloudTrailDataGenerator,
+    cons.VPC_FLOW_TYPE: VPCDataGenerator,
+    cons.KMS_TYPE: KMSDataGenerator,
+    cons.MACIE_TYPE: MacieDataGenerator,
 }
 
 
-def get_data_generator(bucket_type: str) -> DataGenerator:
+def get_data_generator(bucket_type: str, bucket_name: str) -> DataGenerator:
     """Given the bucket type return the correspondant data generator instance.
 
     Args:
         bucket_type (str): Bucket type to match the data generator.
+        bucket_name (str): Bucket name to match in case of custom types.
 
     Returns:
         DataGenerator: Data generator for the given bucket.
     """
+    if bucket_type == cons.CUSTOM_TYPE:
+        bucket_type = bucket_name.split('-')[1]
```

Comment:
What would happen if the `bucket_name` has more than one `-`? Consider using `split("-", 1)` to fix that, unless you   assume there are several `-` and only want the second element between them.

Reply:
Yes, I'm assuming the second case.